lib/deferrable.js

```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+var util = require('util');
+
+
+/**
+ * A collection of properties related to deferrable constraints. It can be used to
+ * make foreign key constraints deferrable and to set the constaints within a
+ * transaction. This is only supported in PostgreSQL.
+ *
+ * The foreign keys can be configured like this. It will create a foreign key
+ * that will check the constraints immediately when the data was inserted.
+ *
+ * ```js
+ * sequelize.define('Model', {
+ *   foreign_id: {
+ *     type: Sequelize.INTEGER,
+ *     references: OtherModel,
+ *     referencesKey: 'id',
+ *     referencesDeferred: Sequelize.Deferrable.INITIALLY_IMMEDIATE
```

Comment:
I personally would prefere deferred instead of referencesDeferred. And I think it would be great to refactor references to use a object as:
```javascript
references: {
    model: OtherModel,
    key: 'id',
    deferred: sequelize.Deferrable.INITIALLY_IMMEDIATE
}
```
This of course would have to be optional though.